mailbox/api/src/main/java/org/apache/james/mailbox/MailboxManager.java

```diff
@@ -385,18 +384,26 @@
      * @return result suitable for the LISTRIGHTS IMAP command
      * @throws UnsupportedRightException
      */
-    MailboxACL.MailboxACLRights[] listRigths(MailboxPath mailboxPath, MailboxACL.MailboxACLEntryKey identifier, MailboxSession session) throws MailboxException;
+    MailboxACL.Rfc4314Rights[] listRigths(MailboxPath mailboxPath, MailboxACL.EntryKey identifier, MailboxSession session) throws MailboxException;
 
     /**
      * Update the Mailbox ACL of the designated mailbox. We can either ADD REPLACE or REMOVE entries.
      *
      * @param mailboxACLCommand Update to perform.
      * @throws UnsupportedRightException
      */
-    void setRights(MailboxPath mailboxPath, MailboxACL.MailboxACLCommand mailboxACLCommand, MailboxSession session) throws MailboxException;
+    void applyRightsCommand(MailboxPath mailboxPath, MailboxACL.ACLCommand mailboxACLCommand, MailboxSession session) throws MailboxException;
 
 
     /**
+     * Reset the Mailbox ACL of the designated mailbox.
+     *
+     * @param mailboxACL New ACL value
+     * @throws UnsupportedRightException
```

Comment:
not sure about this exception

Reply:
in fact it's in the MailboxException hierarchy, so I keep it like this.